adr/ADR-8.md

```diff
@@ -124,9 +124,21 @@ type Status interface {
 	// TTL is how long the bucket keeps values for
 	TTL() time.Duration
 
-	// Keys return a list of all keys in the bucket
+	// Keys return a list of all keys in the bucket.
+	// Historically this method returned a complete slice of all keys in the bucket,
+	// however clients should return interable result.
 	Keys() ([]string, error)
 
+	// KeysWithFilter returns a filtered list of keys in the bucket.
+	// Historically this method returned a complete slice of all keys in the bucket,
+	// however clients should return interable result.
+	KeysWithFilter(filter string) ([]string, error)
```

Comment:
I am really against all these extra methods. Especially against adding both WithFilter and WithFilters versions of these methods, it's not needed at all.

Would be worth adding a watchopt like `FilterKeys(keys ...string)` (for go as this is go here), Keys() already take watchops.